<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>881</width>
    <height>731</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MCModeler[*]</string>
  </property>
  <widget class="QWidget" name="central_widget_">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QScrollArea" name="toolbox_scroll_area_">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="toolbox_scroll_area_contents_">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>76</width>
          <height>684</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QFrame" name="toolbox_frame_">
           <property name="styleSheet">
            <string notr="true">#toolbox_frame_ {
  background-color: palette(base);
}</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QFrame" name="canvas_frame_">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="LevelWidget" name="level_widget_"/>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="level_label_">
            <property name="minimumSize">
             <size>
              <width>26</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="level_slider_">
            <property name="minimum">
             <number>-64</number>
            </property>
            <property name="maximum">
             <number>64</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>8</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar_">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>881</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="file_menu_">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_open_"/>
    <addaction name="separator"/>
    <addaction name="action_save_"/>
    <addaction name="action_save_as_"/>
    <addaction name="separator"/>
    <addaction name="action_quit_"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="action_copy_level_"/>
    <addaction name="action_paste_level_"/>
    <addaction name="separator"/>
    <addaction name="action_extrude_upwards_"/>
    <addaction name="action_extrude_downwards_"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="action_set_template_image_"/>
    <addaction name="action_clear_template_image_"/>
    <addaction name="separator"/>
    <addaction name="action_show_bill_of_materials_"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="action_about_"/>
   </widget>
   <addaction name="file_menu_"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_line_tool_"/>
  </widget>
  <action name="action_open_">
   <property name="text">
    <string>Open…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save_">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_save_as_">
   <property name="text">
    <string>Save As…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_quit_">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_extrude_upwards_">
   <property name="text">
    <string>Extrude Upwards</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Up</string>
   </property>
  </action>
  <action name="action_set_template_image_">
   <property name="text">
    <string>Set Template Image…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="action_clear_template_image_">
   <property name="text">
    <string>Clear Template Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+T</string>
   </property>
  </action>
  <action name="action_extrude_downwards_">
   <property name="text">
    <string>Extrude Downwards</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Down</string>
   </property>
  </action>
  <action name="action_copy_level_">
   <property name="text">
    <string>Copy Level</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="action_paste_level_">
   <property name="text">
    <string>Paste Level</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+V</string>
   </property>
  </action>
  <action name="action_about_">
   <property name="text">
    <string>About MCModeler</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="action_show_bill_of_materials_">
   <property name="text">
    <string>Bill of Materials</string>
   </property>
  </action>
  <action name="action_line_tool_">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Line</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LevelWidget</class>
   <extends>QGraphicsView</extends>
   <header>level_widget.h</header>
   <slots>
    <signal>levelChanged(int)</signal>
    <slot>setLevel(int)</slot>
    <slot>extrudeUpwards()</slot>
    <slot>clearTemplateImage()</slot>
    <slot>extrudeDownwards()</slot>
    <slot>copyLevel()</slot>
    <slot>pasteLevel()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>level_slider_</sender>
   <signal>valueChanged(int)</signal>
   <receiver>level_label_</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>866</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>865</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>level_slider_</sender>
   <signal>valueChanged(int)</signal>
   <receiver>level_widget_</receiver>
   <slot>setLevel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>866</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>607</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_quit_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_open_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_as_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_extrude_upwards_</sender>
   <signal>triggered()</signal>
   <receiver>level_widget_</receiver>
   <slot>extrudeUpwards()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>level_widget_</sender>
   <signal>levelChanged(int)</signal>
   <receiver>level_slider_</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>655</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>855</x>
     <y>412</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_set_template_image_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setTemplateImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_clear_template_image_</sender>
   <signal>triggered()</signal>
   <receiver>level_widget_</receiver>
   <slot>clearTemplateImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_extrude_downwards_</sender>
   <signal>triggered()</signal>
   <receiver>level_widget_</receiver>
   <slot>extrudeDownwards()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_copy_level_</sender>
   <signal>triggered()</signal>
   <receiver>level_widget_</receiver>
   <slot>copyLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_paste_level_</sender>
   <signal>triggered()</signal>
   <receiver>level_widget_</receiver>
   <slot>pasteLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_about_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_show_bill_of_materials_</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showBillOfMaterials()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>quit()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>open()</slot>
  <slot>setTemplateImage()</slot>
  <slot>about()</slot>
  <slot>showBillOfMaterials()</slot>
 </slots>
</ui>
